File: 3rd-semester/object-oriented-programming/containers/csharp/src/Models/RefrigeratedContainer.cs
namespace ShipCargoSystem.Models;

/// <summary>
/// A refrigerated shipping container charged based on power consumption.
/// <para>Cost formula: <c>powerKw × 2000</c></para>
/// </summary>
public sealed class RefrigeratedContainer : Container
{
    private const double RatePerKw = 2000.0;

    /// <summary>Gets the power consumption in kilowatts.</summary>
    public double PowerKw { get; }

    /// <summary>Initialises a RefrigeratedContainer.</summary>
    /// <param name="code">Unique container identifier.</param>
    /// <param name="destination">Destination port or city.</param>
    /// <param name="powerKw">Power consumption in kilowatts.</param>
    public RefrigeratedContainer(string code, string destination, double powerKw)
        : base(code, destination) => PowerKw = powerKw;

    /// <inheritdoc/>
    public override double GetCost() => RatePerKw * PowerKw;

    /// <inheritdoc/>
    public override string ToString() =>
        $"RefrigeratedContainer[code={Code}, destination={Destination}, power={PowerKw:F1} kW, cost={GetCost():F2}]";
}

File: 3rd-semester/object-oriented-programming/containers/java/src/main/java/com/university/oop/containers/model/RefrigeratedContainer.java
package com.university.oop.containers.model;

/**
 * A refrigerated shipping container charged based on power consumption.
 *
 * <p>Cost formula: {@code powerKw × 2000}</p>
 */
public final class RefrigeratedContainer extends Container {

    /** Cost rate per kilowatt of power. */
    private static final double RATE_PER_KW = 2000.0;

    private final double powerKw;

    /**
     * Constructs a RefrigeratedContainer.
     *
     * @param code        unique container identifier
     * @param destination destination port or city
     * @param powerKw     power consumption in kilowatts
     */
    public RefrigeratedContainer(String code, String destination, double powerKw) {
        super(code, destination);
        this.powerKw = powerKw;
    }

    /** @return power consumption in kilowatts */
    public double getPowerKw() { return powerKw; }

    @Override
    public double getCost() { return RATE_PER_KW * powerKw; }

    @Override
    public String toString() {
        return String.format("RefrigeratedContainer[code=%s, destination=%s, power=%.1f kW, cost=%.2f]",
                getCode(), getDestination(), powerKw, getCost());
    }
}
